mode2.lua

```diff
@@ -961,43 +978,63 @@ function courseplay:unload_combine(vehicle, dt)
 		if distance_to_wp > vehicle.cp.shortestDistToWp and distance_to_wp < 3 then
 			distToChange = distance_to_wp + 1
 		end
-		if distance_to_wp < distToChange then
-			--[[if vehicle.cp.mode2nextState == 81 then
-				if vehicle.cp.activeCombine ~= nil then
-					courseplay:unregisterFromCombine(vehicle, vehicle.cp.activeCombine)
-				end
-			end]]
 
+		if distance_to_wp < distToChange then
+      -- Switching to next waypoint
 			vehicle.cp.shortestDistToWp = nil
 			if #(vehicle.cp.nextTargets) > 0 then
-				-- courseplay:setModeState(vehicle, 5);
-				courseplay:setCurrentTargetFromList(vehicle, 1);
+        -- still have waypoints left
+        -- if we are chasing our combine, check if we are close enough now (this can happen when
+        -- it has moved since we set our course to it and we are now driving close by) In this case.
+        -- abort the course and catch it.
+        local continueCourse = true
+        if vehicle.cp.mode2nextState == STATE_DRIVE_TO_COMBINE then
+          -- how far it is then?
+	        local combine = vehicle.cp.activeCombine or vehicle.cp.lastActiveCombine;
+          if combine then
+            local distanceToCombine = courseplay:distanceToObject( vehicle, combine )
+            if distanceToCombine < 50 then
```

Comment:
I noticed that even when I stop following waypoints soon enough when heading the opposite direction of the combine, the tractor will keep running for a while before it turns back and actually starts following it. Not sure where this is implemented and how to avoid it, it would be better to just turn immediately.